llvm/lib/Target/AIE/AIEBasePipelinerLoopInfo.cpp

```diff
@@ -664,6 +669,8 @@ class ZeroOverheadLoop : public AIEBasePipelinerLoopInfo {
       SmallVectorImpl<MachineOperand> &Cond) override;
 
   bool canAcceptII(SMSchedule &SMS) override;
+
+  bool shouldUseSchedule(SwingSchedulerDAG &SSD, SMSchedule &SMS) override;
```

Comment:
We should probably update `canAcceptII` as well. Typically, this will increase the II until we have a low enough stage count. So I think that if we find a schedule with a low II and high stage count, we should immediately refuse it sop the post-pipeliner can pick it; and not increase the II.

Reply:
That's an interface change I think? it should be able to say yes, higherII and stop. That would definitely save some time.

Reply:
I guess we would accept the II if we are confident we can pick up the loop in the post-pipeliner. And for those loops, `shouldUseSchedule` would return `false`.

Reply:
Somehow that makes sense, except for the name of canAcceptII. I will add a fat comment.